Tools/CMake/AMReXTargetHelpers.cmake

```
@@ -114,12 +114,12 @@ endfunction ()
 # is compatible with amrex CUDA build.
 #
 function (setup_target_for_cuda_compilation _target)
-   # separable compilation:
-   #   mainly due to amrex::Random which uses global device variables
-   set_target_properties( ${_target}
-      PROPERTIES
-      CUDA_SEPARABLE_COMPILATION ON      # This adds -dc
-      )
+   if (AMReX_GPU_RDC)
+      set_target_properties( ${_target}
+         PROPERTIES
+         CUDA_SEPARABLE_COMPILATION ON      # This adds -dc
+         )
+   endif ()
```

With the default set above, we can do this:
```suggestion
   set_target_properties( ${_target}
      PROPERTIES
      CUDA_SEPARABLE_COMPILATION ${AMReX_GPU_RDC}      # This adds -dc
   )
```